### ISSUE TITLE
Split word on apostrophe when calculating word count

### ISSUE THREAD

**Issue by diedexx** [Yoast/YoastSEO.js#438]
At the moment the word count check checks for words by splitting on whitespace ([here](https://github.com/Yoast/YoastSEO.js/blob/develop/js/stringProcessing/countWords.js#L18))

This sentence: "I'm glad you're ok." contains 4 words according to Yoast SEO. however, there actually are 6 words: "I am glad you are ok.".

We should split on apostrophes when counting words.

### Technical info
* Wordpress version: 4.4.2
* Yoast SEO version: 3.2 beta 4

**Comment by GSAdev** [comment]
Not all words with an apostrophe should be splitted. The apostrophe is also used as a marker of e.g. a possessive relationship: For example: My wife's car. In Danish the apostrophe _can_ be used as a marker of contraction, but most often it is a marker of a possessive relationship. Sometimes followed by a 's', sometimes followed by nothing. I would suspect that one needs to parse a phrase in order to determine what function the apostrophe has.

**Comment by atimmer** [comment]
@IreneStr Can you comment on this?

**Comment by IreneStr** [comment]
Whether 'I'm' and 'you're' are one or two words depends heavily on your definition of 'word'. For both cases there are many convincing arguments. 

However, like @GSAdev wrote correctly, it is dangerous to split on every apostrophe. In order to split correctly, we'd have to work with wordlists to check against the word part before the apostrophe to make sure we have to split or not. For example: in English ''s' is a verb (an therefore possibly a word) after personal pronouns, but it indicates possession after names and nouns. 

Moreover, I can imagine that whether or not clitics (the parts after a apostrophe) are viewed as separate words is highly language specific. Not all languages use word boundaries the same way. Splitting on apostrophes would possibly result in a worse instead of a better analysis. 

In conclusion, (for the time being) I'd recommend not splitting on apostrophes, because it could possibly break more than it would fix and the analysis would heavily depend on word lists.
